sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/dao/PartitionMetadataDao.java

```diff
@@ -96,6 +96,30 @@ public boolean tableExists() {
     }
   }
 
+  /**
+   * Finds all indexes for the metadata table.
+   *
+   * @return a list of index names for the metadata table.
+   */
+  public List<String> findAllTableIndexes() {
+    final String indexesStmt =
```

Comment:
We probably need to do this for both dialects

Reply:
This works for both dialects (tested manually)

Reply:
Sg, a small nit would be to add table_schema filtering - '' for GoogleSQL and 'public' for Postgres. That way this query will behave expectedly for dbs with named schemas (which we currently don't support)

Reply:
Done